packages/components/src/color-picker/inputs.js

```diff
@@ -22,159 +21,148 @@ import VisuallyHidden from '../visually-hidden';
 import { isValidHex } from './utils';
 
 /* Wrapper for TextControl, only used to handle intermediate state while typing. */
-export class Input extends Component {
-	constructor() {
-		super( ...arguments );
-		this.handleBlur = this.handleBlur.bind( this );
-		this.handleChange = this.handleChange.bind( this );
-		this.handleKeyDown = this.handleKeyDown.bind( this );
-	}
 
-	handleBlur() {
-		const { value, valueKey, onChange, source } = this.props;
+export function Input( {
+	label,
+	value,
+	valueKey,
+	onChange,
+	source,
+	...props
+} ) {
+	const handleBlur = () => {
 		onChange( {
 			source,
 			state: 'commit',
 			value,
 			valueKey,
 		} );
-	}
+	};
 
-	handleChange( value ) {
-		const { valueKey, onChange, source } = this.props;
-		if ( value.length > 4 && isValidHex( value ) ) {
+	const handleChange = ( newValue ) => {
+		if ( newValue.length > 4 && isValidHex( newValue ) ) {
 			onChange( {
 				source,
 				state: 'commit',
-				value,
+				value: newValue,
 				valueKey,
 			} );
 		} else {
 			onChange( {
 				source,
 				state: 'draft',
-				value,
+				value: newValue,
 				valueKey,
 			} );
 		}
-	}
+	};
 
-	handleKeyDown( { keyCode } ) {
+	const handleKeyDown = ( { keyCode } ) => {
 		if ( keyCode !== ENTER && keyCode !== UP && keyCode !== DOWN ) {
 			return;
 		}
-		const { value, valueKey, onChange, source } = this.props;
 		onChange( {
 			source,
 			state: 'commit',
 			value,
 			valueKey,
 		} );
-	}
+	};
 
-	render() {
-		const { label, value, ...props } = this.props;
-		return (
-			<TextControl
-				className="components-color-picker__inputs-field"
-				label={ label }
-				value={ value }
-				onChange={ ( newValue ) => this.handleChange( newValue ) }
-				onBlur={ this.handleBlur }
-				onKeyDown={ this.handleKeyDown }
-				{ ...omit( props, [ 'onChange', 'valueKey', 'source' ] ) }
-			/>
-		);
-	}
+	return (
+		<TextControl
+			className="components-color-picker__inputs-field"
+			label={ label }
+			value={ value }
+			onChange={ ( newValue ) => handleChange( newValue ) }
+			onBlur={ handleBlur }
+			onKeyDown={ handleKeyDown }
+			{ ...omit( props, [ 'onChange', 'valueKey', 'source' ] ) }
```

Comment:
There's no need to omit these anymore.

Reply:
I removed `omit` in daddou-ma/gutenberg@adaa488.